crates/dojo-lang/src/compiler.rs

```diff
@@ -69,30 +69,73 @@ pub struct DojoCompiler;
 #[derive(Debug, Default, Serialize, Deserialize)]
 #[serde(rename_all = "kebab-case")]
 pub struct Props {
-    pub build_external_contracts: Option<Vec<ContractSelector>>,
+    pub build_external_contracts: Option<Vec<BaseContractSelector>>,
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub struct ContractSelector(String);
+pub struct BaseContractSelector(String);
 
-impl ContractSelector {
+impl BaseContractSelector {
     fn package(&self) -> PackageName {
         let parts = self.0.split_once(CAIRO_PATH_SEPARATOR).unwrap_or((self.0.as_str(), ""));
         PackageName::new(parts.0)
     }
 
-    /// Returns the path with the model name in snake case.
+    // Check if the contract selector matches with the provided contract path.
+    // Note that this contract path uses the Dojo format in Snaked case, which is :
+    //     `<path>::<namespace><naming::DOJO_CONTRACT_SEPARATOR><name>`.
+    // That means, to match the provided contract path, the contract selector path and name
+    // must match with this contract path `path` and `name`.
+    fn match_contract_path(&self, contract_path: String) -> bool {
+        let (source_path, source_name) =
+            contract_path.rsplit_once(CAIRO_PATH_SEPARATOR).unwrap_or(("", &contract_path));
+        let (_, source_name) =
+            source_name.rsplit_once(naming::DOJO_CONTRACT_SEPARATOR).unwrap_or(("", source_name));
+
+        let (path, name) = self.0.rsplit_once(CAIRO_PATH_SEPARATOR).unwrap_or(("", &self.0));
+        let name = name.to_case(Case::Snake);
+
+        source_path == path && source_name == name
+    }
```

Comment:
We're not including the namespace in the comparison here, may we have conflicts at some point?

Reply:
In fact, this `match_contract_path` function is used for external contracts only (the ones specified in `Scarb.toml`). I don't know if these external contracts can have a namespace in their names 🤔 
But, for the `spawn-and-move` example, we have 2 external contracts without namespace (for example, the contract selector `bestiary::RiverSkale` must be matched with the contract path `bestiary::dojo_examples_foes__river_skale`).

So, I propose to compare namespaces if and only if the contract selector contains a namespace. But, in case of namespace remapping, is it possible to have an external contract with a namespace in its name remapped to another namespace ?